src/Http/Controllers/Character/MarketController.php

```diff
@@ -57,18 +58,37 @@ public function getMarket(int $character_id)
     public function getMarketData(int $character_id)
     {
 
-        $orders = $this->getCharacterMarketOrders($character_id, false);
+        if (! request()->has('all_linked_characters'))
+            return abort(500);
+
+        if (request('all_linked_characters') === 'false')
+            $character_ids = collect($character_id);
+
+        $user_group = User::find($character_id)->group->users
+            ->filter(function ($user) {
+                return $user->name !== 'admin' && $user->id !== 1;
+            })
+            ->pluck('id');
+
+        if (request('all_linked_characters') === 'true')
+            $character_ids = $user_group;
+
+        $orders = $this->getCharacterMarketOrders($character_ids);
 
         return DataTables::of($orders)
             ->addColumn('bs', function ($row) {
 
-                return view('web::partials.marketbuysell', compact('row'))
-                    ->render();
+                if ($row->is_buy_order)
+                    return '<span class="text-red">Buy</span>';
+
+                return '<span class="text-green">Sell</span>';
             })
             ->addColumn('vol', function ($row) {
 
-                return view('web::partials.marketvolume', compact('row'))
-                    ->render();
+                if ($row->is_buy_order)
+                    return number($row->volume_total, 0);
```

Comment:
I may agree with the number thing, but can you keep it into the blade ?
Also, with the asynchronous working, shouldn't the number feature be moved into a datatable render ?

Reply:
About the numbers: https://github.com/eveseat/web/blame/a1274fa520c2925b3fb5396fa2864d8d0d8fa1f9/src/Http/Controllers/Character/MarketController.php#L75 
That's the way it has been used for quite some time and the way the helper function works, this seems to be right. Doing this asynchronously now, does not have anything to do how the number format is rendered. Either you render it in the controller or by the client. However if you'd let the client render it. We might need to move the whole helper to a js version. This is comparable to the human_readable js helper: 
https://github.com/eveseat/web/blob/master/src/resources/assets/js/seat.js#L67-L75

i have the feeling, editing the column in the asynchronous controller is the right place to do so. And i would suggest keep this as is. 

The partial blade for `<span class="text-red">Buy</span>` i have restored, this i understand